src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
The rule is basically that the change attr must be different from when you last queried for it iff _something_ changed. That something could be an inode metadata or data change. The application is not allowed to infer anything from the value itself, however. It can only test it for equivalence with an earlier value. So, it doesn't really matter how much the thing changes as long as the above rule is followed.